ops/model.py

```diff
@@ -217,16 +219,23 @@ def __getitem__(self, relation_name):
         if relation_list is None:
             relation_list = self._data[relation_name] = []
             for relation_id in self._backend.relation_ids(relation_name):
-                relation_list.append(Relation(relation_name, relation_id, self._our_unit, self._backend, self._cache))
+                relation_role = self._relations_meta[relation_name].role
```

Comment:
You don't have to look this up in the loop, right? It doesn't change based on the relation-id.

Reply:
Right, thanks for noticing.